platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/AppCommunicationComponentWiring.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.wiring.components;
+
+import com.swirlds.common.wiring.schedulers.TaskScheduler;
+import com.swirlds.common.wiring.wires.input.BindableInputWire;
+import com.swirlds.common.wiring.wires.input.InputWire;
+import com.swirlds.platform.components.appcomm.AppCommunicationComponent;
+import com.swirlds.platform.state.signed.ReservedSignedState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Wiring for the {@link AppCommunicationComponent}.
+ *
+ * @param appCommunicationComponentInputWire  the input wire for app communication component
+ */
+public record AppCommunicationComponentWiring(
+        @NonNull InputWire<ReservedSignedState> appCommunicationComponentInputWire) {
```

Comment:
The name `appCommunicationComponentInputWire` doesn't describe what typ eof object is being input here. I think something like `signedStateInput` would be better

Reply:
By the way, why do we mostly name the input wire variables *Input? Considering it's a wire and not the input itself, I'm curious to know how this convention came by?

Reply:
There's no strong requirement to exclude the word "wire", it's just something I've been doing since some of the member names were exceeding a reasonable length. However, I don't thing calling an input wire an "input" is a misnomer: "input" can refer to *what* is being put in, or *where* it is being put in.